src/components/wallet/methods/depositAsset.ts

```diff
@@ -30,25 +26,14 @@ export default async function depositAsset(
       this.account.publicKey
     )
 
-    const balances = loGet(this.account, 'state.balances')
-    const hasCurrency = loFindIndex(balances, {
-      asset_code,
-      asset_issuer,
-    })
+    this.logger.request('Fetch issuer account from Horizon', asset_issuer)
+    const issuerAccount = await this.server.loadAccount(asset_issuer)
+    this.logger.response('Fetch issuer account from Horizon', issuerAccount)
+    const homeDomain = issuerAccount.home_domain
+    if (!homeDomain) throw "Couldn't find a home_domain on the assets issuer"
 
-    if (hasCurrency === -1)
-      await this.trustAsset(asset_code, asset_issuer, pincode_stretched)
-
-    const server = this.server as Server
-    const issuerAccount = await server.loadAccount(asset_issuer)
-    const homeDomain: string = (issuerAccount as any).home_domain
```

Comment:
We can remove a good deal of the Typescript type casting as we now have proper types in `this`, due to the 'fake this' feature of typescript.  We pass a value called `this` to the function, and typescript will use that type to infer the true type of `this`.